### PR TITLE
channels_last support to batchnorm layer

### PR DESCRIPTION
**Pull request by avitalsh**
Added channels_last (NHWC format) support to batch norm layer and updated the unitest accordingly. 
Channels_first remains the default.

### CODE REVIEW

File: tests/test_batchnorm.py [hunk 1]
```diff
@@ -14,50 +14,86 @@ def setUp(self):
...
+          sess.run(tf.global_variables_initializer())
 
-        out_tensorflow = sess.run(batchnorm_out_tf)
+          out_tensorflow = sess.run(batchnorm_out_tf)
 
-        np.testing.assert_array_almost_equal(
+          np.testing.assert_array_almost_equal(
             out_pond, out_tensorflow, decimal=1)
 
+    def test_channels_first():
```
Comment (jvmancuso) [inline tests/test_batchnorm.py]: can you move these two out into their own methods on TestBatchnorm? that way if one of them fails in the future it's a bit more informative for the person running this file

Comment (jvmancuso) [reply tests/test_batchnorm.py]: you can also look into using `pytest.mark.parametrize` for this if you're feeling ambitious! (but not necessary for merging)

Comment (avitalsh) [reply tests/test_batchnorm.py]: @jvmancuso Thanks for the review. I separated the tests. Tried looking at ```pytest.mark.parametrize``` but it seems to be not working great with ```unittest.TestCase``` subclasses -
https://stackoverflow.com/questions/35561843/py-test-parametrize-test-cases-from-classes
https://stackoverflow.com/questions/18182251/does-pytest-parametrized-test-work-with-unittest-class-based-tests

Comment (jvmancuso) [reply tests/test_batchnorm.py]: I see, in that case parameterization is probably beyond the scope of this PR. LGTM!
